### PR TITLE
Groundwork for new publishing API

### PR DESCRIPTION
**Pull request by excelsior**

### CODE REVIEW

File: db/migrate/20171101194708_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class CreateUsers < ActiveRecord::Migration
+  def change
+    create_table :users do |t|
+      t.references :admin, foreign_key: true
+      t.string :email, null: false
```
Comment (andersoncardoso) [inline db/migrate/20171101194708_create_users.rb]: `unique: true`, since we are validating for it.

File: app/models/key_pair.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# RSA key pair used for server-side signing
+class KeyPair < ActiveRecord::Base
+  SECRET_KEY = ENV.fetch('ENCRYPTED_PRIVATE_KEY_SECRET')
```
Comment (andersoncardoso) [inline app/models/key_pair.rb]: this key uses any special format, or it's just an arbitrary string? 
We can add a default (as documentation, and for dev) on the `.env` file.

File: app/models/key_pair.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  private
+
+  # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+  def generate_keys
+    dir_path = Rails.root.join('tmp', 'keys', organization_publisher_id.to_s)
+    FileUtils.mkdir_p(dir_path)
+
+    pem_path = dir_path + 'id_rsa.pem'
```
[the remark this answers is not in the document]
Comment (excelsior) [reply app/models/key_pair.rb]: `Rails.root` returns a `Pathname` object. I made `MR.root_path` behave the same way. And `Pathname#+` takes care of the trailing slash when appending fragments, too. Let me know if you see any downsides of this change.

[the remark this answers is not in the document]
Comment (andersoncardoso) [reply app/models/key_pair.rb]: 👍

**Pull request merged**